src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+import usersFromServer from '../../api/users';
+import { User } from '../../types/User';
+import { getUserById } from '../../services/user';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitile] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleSetTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitile(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleSetUserIdChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setUserId(0);
+    setTitile('');
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      title,
+      userId,
+      user: getUserById(userId),
+      completed: false,
+    });
+
+    reset();
+  };
+
+  return (
+    <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
+      <div className="field">
+        <input
+          type="text"
+          data-cy="titleInput"
+          value={title}
+          placeholder="Enter a title"
+          onChange={handleSetTitleChange}
+        />
+        {hasTitleError && <span className="error">Please enter a title</span>}
```

Comment:
use it for all such cases(more readable)
```suggestion
        {hasTitleError && (
          <span className="error">
            Please enter a title
          </span>
        )}
```

Reply:
My linter says that this is not correct
<img width="323" alt="image" src="https://github.com/user-attachments/assets/e507b276-2b16-4b4f-9205-e887ede323c6">